### ISSUE TITLE
Dialog.create() no longer supports registered components

### ISSUE THREAD

**Issue by steersbob** [quasarframework/quasar#8885]
**Describe the bug**
In Quasar v1, Dialog.create() accepts a string component name of an earlier registered component.
In Quasar v2, doing the same results in an error.

Example call: `Dialog.create({component: 'CustomDialog'})`

Error:
```
runtime-core.esm-bundler.js?9e79:217 Uncaught TypeError: dialogRef.value.show is not a function
    at Object.__webpack_exports__.default [as dialog] (global-dialog.js?0bab:131)
    at badDialog (Index.vue?8b24:27)
    at callWithErrorHandling (runtime-core.esm-bundler.js?9e79:154)
    at callWithAsyncErrorHandling (runtime-core.esm-bundler.js?9e79:163)
    at emit (runtime-core.esm-bundler.js?9e79:642)
    at eval (runtime-core.esm-bundler.js?9e79:6689)
    at onClick (QBtn.js?9c40:143)
    at callWithErrorHandling (runtime-core.esm-bundler.js?9e79:154)
    at callWithAsyncErrorHandling (runtime-core.esm-bundler.js?9e79:163)
    at HTMLButtonElement.invoker (runtime-dom.esm-bundler.js?8886:333)
```

**Codepen/jsFiddle/Codesandbox (required)**
https://github.com/steersbob/dialog-bug

This contains the output from `quasar create dialog-bug --branch next`, with changes to some files:
- src/components/CustomDialog.vue -> copy/paste of the example at https://next.quasar.dev/vue-composables/use-dialog-plugin-component
- quasar.conf.js -> added the `Dialog` plugin
- src/boot/register.js -> register CustomDialog: `app.component('CustomDialog', CustomDialog)`
- src/pages/Index.vue -> add "Good" and "Bad" buttons
```
    function badDialog() {
      $q.dialog({component: 'CustomDialog'})
    }
    function goodDialog() {
      $q.dialog({component: CustomDialog})
    }
```

**To Reproduce**
1. Run `quasar dev`
2. Open localhost:8080 and dev tools
3. Click "Good" -> a barebones dialog is spawned
4. Click "Bad" -> Nothing spawns, an error is printed to console

**Expected behavior**
"Good" and "Bad" buttons to have identical behavior

**Screenshots**

**Platform (please complete the following information):**
Quasar Version: 2
@quasar/app Version: 3.0.0-beta.1
Quasar mode:
  - [X] SPA
  - [ ] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

Tested on:
  - [X] SPA
  - [ ] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

OS: Ubuntu 20.04
Node: 14
NPM: 6.14
Yarn: 1.22
Browsers: Chrome / Firefox
iOS: N/A
Android: N/A
Electron: N/A

**Additional context**
I dug around in global-dialog.js, and my own theory is that the QGlobalDialog child app does not inherit registered components from the parent app, but I haven't confirmed this.

**Comment by rstoenescu** [comment]
Thanks for also supplying a PR!
Will be included in Quasar v2.0.0-beta.13.

**Comment by omartb89** [comment]
someone got a codepen with a funtional example?
